src/hooks/useRestoreOriginalUri.tsx

```diff
@@ -0,0 +1,52 @@
+/*!
+ * Copyright (c) 2017-Present, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import * as React from 'react';
+import { OktaAuth } from '@okta/okta-auth-js';
+import { RestoreOriginalUriFunction } from '../types';
+
+const useRestoreOriginalUri = (
+  oktaAuth: OktaAuth,
+  restoreOriginalUri: RestoreOriginalUriFunction,
+): null => {
+  // We don't want any side effects (excess render, effect run) on `restoreOriginalUri` change,
+  //  so keep and use the latest value from the ref
+  const restoreOriginalUriRef = React.useRef(restoreOriginalUri);
```

Comment:
Why is a ref needed here?

Reply:
It's a safe way to eliminate `Two custom restoreOriginalUri callbacks are detected` warning issues.
When another instance  of `restoreOriginalUri` is passed to `Security`, the effect would not be triggered.
Using [empty dependencies array](https://github.com/okta/okta-react/blob/master/src/Security.tsx#L63) also works, but gives lint error after adding `react-hooks/rules-of-hooks` to ESlint config